core/src/Kokkos_View.hpp

```diff
@@ -97,26 +97,53 @@ constexpr KOKKOS_INLINE_FUNCTION std::size_t count_valid_integers(
          (i6 != KOKKOS_INVALID_INDEX) + (i7 != KOKKOS_INVALID_INDEX);
 }
 
-KOKKOS_INLINE_FUNCTION
-void runtime_check_rank(const size_t rank, const size_t dyn_rank,
-                        const bool is_void_spec, const size_t i0,
-                        const size_t i1, const size_t i2, const size_t i3,
-                        const size_t i4, const size_t i5, const size_t i6,
-                        const size_t i7, const std::string& label) {
+template <typename View>
+KOKKOS_INLINE_FUNCTION void runtime_check_rank(
+    const View&, const size_t rank, const size_t dyn_rank,
+    const bool is_void_spec, const size_t i0, const size_t i1, const size_t i2,
+    const size_t i3, const size_t i4, const size_t i5, const size_t i6,
+    const size_t i7, const std::string& label) {
```

Comment:
I know you did not change that part but how does that even work a `std::string` argument on the device?

Reply:
I guess the compiler ignores `label` on the device since it's a reference and we don't do any operations with it?